<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>585</width>
    <height>659</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QFrame" name="frame_2">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QPushButton" name="pushButton_31">
         <property name="text">
          <string>测试用文件</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit"/>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>字节序</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QRadioButton" name="rbt_byte_order_littleEndian">
            <property name="text">
             <string>LittleEndian</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="rbt_byt_order_bigEndian">
            <property name="text">
             <string>BigEndian</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>浮点数精度</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QRadioButton" name="rbt_float_precision">
            <property name="text">
             <string>SinglePrecision</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="rbt_double_precision">
            <property name="text">
             <string>DoublePrecision</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="title">
       <string>与编码数据序列化方式读写测试</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="7" column="2">
        <widget class="QPushButton" name="pushButton_20">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="3" column="3">
        <widget class="QPushButton" name="pushButton_8">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>float</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="label_14">
         <property name="contextMenuPolicy">
          <enum>Qt::CustomContextMenu</enum>
         </property>
         <property name="text">
          <string>qfloat16</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QLineEdit" name="lineEdit_2"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>qint16(-32768-32767))</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QSpinBox" name="spinBox_3"/>
       </item>
       <item row="9" column="0">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>double</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="11" column="1">
        <widget class="QLineEdit" name="lineEdit_15"/>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_8">
         <property name="contextMenuPolicy">
          <enum>Qt::CustomContextMenu</enum>
         </property>
         <property name="text">
          <string>Bool</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="12" column="0">
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>QFont</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="12" column="4">
        <widget class="QLineEdit" name="lineEdit_11"/>
       </item>
       <item row="5" column="4">
        <widget class="QLineEdit" name="lineEdit_7"/>
       </item>
       <item row="0" column="3">
        <widget class="QPushButton" name="pushButton_2">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QLineEdit" name="lineEdit_3"/>
       </item>
       <item row="6" column="1">
        <widget class="QCheckBox" name="checkBox">
         <property name="text">
          <string>布尔值</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>qint64</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="12" column="3">
        <widget class="QPushButton" name="pushButton_28">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="2" column="4">
        <widget class="QLineEdit" name="lineEdit_4"/>
       </item>
       <item row="8" column="1">
        <widget class="QDoubleSpinBox" name="doubleSpinBox_12"/>
       </item>
       <item row="8" column="2">
        <widget class="QPushButton" name="pushButton_21">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="10" column="4">
        <widget class="QLineEdit" name="lineEdit_10"/>
       </item>
       <item row="6" column="3">
        <widget class="QPushButton" name="pushButton_13">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="spinBox"/>
       </item>
       <item row="5" column="1">
        <widget class="QSpinBox" name="spinBox_6"/>
       </item>
       <item row="5" column="3">
        <widget class="QPushButton" name="pushButton_12">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="8" column="3">
        <widget class="QPushButton" name="pushButton_24">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="13" column="2">
        <widget class="QPushButton" name="pushButton_16">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="9" column="4">
        <widget class="QLineEdit" name="lineEdit_8"/>
       </item>
       <item row="3" column="4">
        <widget class="QLineEdit" name="lineEdit_5"/>
       </item>
       <item row="6" column="2">
        <widget class="QPushButton" name="pushButton_14">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="11" column="2">
        <widget class="QPushButton" name="pushButton_18">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_2">
         <property name="contextMenuPolicy">
          <enum>Qt::CustomContextMenu</enum>
         </property>
         <property name="text">
          <string>qint8(-128-127)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="11" column="3">
        <widget class="QPushButton" name="pushButton_23">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="10" column="1">
        <widget class="QLineEdit" name="lineEdit_14"/>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>quint8(0-255)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="7" column="4">
        <widget class="QLineEdit" name="lineEdit_12"/>
       </item>
       <item row="4" column="2">
        <widget class="QPushButton" name="pushButton_9">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="10" column="3">
        <widget class="QPushButton" name="pushButton_27">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QPushButton" name="pushButton">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QPushButton" name="pushButton_7">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="spinBox_2"/>
       </item>
       <item row="3" column="1">
        <widget class="QSpinBox" name="spinBox_4"/>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="pushButton_3">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="10" column="0">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>char*</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="5" column="2">
        <widget class="QPushButton" name="pushButton_11">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="8" column="4">
        <widget class="QLineEdit" name="lineEdit_9"/>
       </item>
       <item row="9" column="1">
        <widget class="QDoubleSpinBox" name="doubleSpinBox_11"/>
       </item>
       <item row="6" column="4">
        <widget class="QCheckBox" name="checkBox_2">
         <property name="text">
          <string>布尔值</string>
         </property>
        </widget>
       </item>
       <item row="13" column="3">
        <widget class="QPushButton" name="pushButton_17">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QDoubleSpinBox" name="doubleSpinBox_9"/>
       </item>
       <item row="13" column="0">
        <widget class="QLabel" name="label_11">
         <property name="contextMenuPolicy">
          <enum>Qt::CustomContextMenu</enum>
         </property>
         <property name="text">
          <string>QColor</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="9" column="3">
        <widget class="QPushButton" name="pushButton_15">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="11" column="4">
        <widget class="QLineEdit" name="lineEdit_13"/>
       </item>
       <item row="1" column="3">
        <widget class="QPushButton" name="pushButton_4">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QPushButton" name="pushButton_6">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="4" column="4">
        <widget class="QLineEdit" name="lineEdit_6"/>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>quint16(0-65536)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="12" column="2">
        <widget class="QPushButton" name="pushButton_26">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="4" column="3">
        <widget class="QPushButton" name="pushButton_10">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="9" column="2">
        <widget class="QPushButton" name="pushButton_22">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QSpinBox" name="spinBox_5"/>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>qint32</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="7" column="3">
        <widget class="QPushButton" name="pushButton_19">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPushButton" name="pushButton_5">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="11" column="0">
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>QString</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="10" column="2">
        <widget class="QPushButton" name="pushButton_25">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="12" column="1">
        <widget class="QPushButton" name="pushButton_29">
         <property name="text">
          <string>选择字体</string>
         </property>
        </widget>
       </item>
       <item row="13" column="1">
        <widget class="QPushButton" name="pushButton_30">
         <property name="text">
          <string>选择颜色</string>
         </property>
        </widget>
       </item>
       <item row="13" column="4">
        <widget class="QLineEdit" name="lineEdit_16"/>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="act_clear_edit_widget"/>
   <addaction name="act_write_file_continuous"/>
   <addaction name="act_read_file__continuous"/>
   <addaction name="act_exit"/>
  </widget>
  <action name="act_clear_edit_widget">
   <property name="text">
    <string>清空编辑框</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="act_write_file_continuous">
   <property name="text">
    <string>连续写入文件</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="act_read_file__continuous">
   <property name="text">
    <string>连续从文件读取</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="act_exit">
   <property name="text">
    <string>退出</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
